### ISSUE TITLE
Pokemons with TTL of 45min or 30min are found only after their TTL hits 15min

### ISSUE THREAD

**Issue by MzHub** [PokemonGoMap/PokemonGo-Map#162]
Glossary: TTL = time to live, as in how long until the pokemon despawns.

**Current behaviour**
Example:
A 45min spawn point becomes active at 00:05:00.
I can now see the pokemon in my nearby and catch it.
Forcing a scan at this location shows no pokemon.
Waiting until 00:35:00.
After this time the pokemon can be seen when the spot is scanned.
TTL shows less than 15 min.

**Expected behaviour**
Map shows the pokemon after 00:05:00, when it spawns.

**Possible solution**
~~Something to do with the "since_timestamp_ms"?~~ EDIT: Likely not.
I don't know really, although it seems as if someone had made the assumption that spawns are always 15min TTL spawns.

**Context**
The effect from this is that a portion of the pokemon that have spawned in the wild are not visible in the map until they only have 15 minutes before despawning. Also having 30min and 45min spawns visible would make it easier to reach rare spawns that happen to have longer TTL and are far enough away that 15 minutes is too short.

**Environment**
According to the database I've collected with different versions of the map, this has existed pretty much always. My database does not have a single spawn logged that had a TTL above 15 minutes, which should have happened at least sometimes.
No specific environment.

EDIT: I've also checked that the actual API response contains no wild_pokemons entries when the spawn should have activated, even though the pokemon is catchable.

**Comment by BaIthamel** [comment]
Have you tried using the pogoapi without the map to confirm it's an issue with the map?

**Comment by thesandvichisaliar** [comment]
The pokemon are being scanned and logged in the database just fine. The issue is that pokemon with > 15 minutes until disappear_time return with a negative time, e.g. 2016-07-18 21:53:35.231000. And since the map removes pokemon from the display when they pass their disappear_time, you don't see them until you see the valid disappear_time sent when they were within 15 minutes. We could hard code a time threshold into the map to show disappear_times in a certain past range that simply display with "15+ minutes left" since you can't initially tell whether it is a 30 or 45 minute spawn.

**Comment by MzHub** [comment]
Yeah, you're right, I see it now as well. They always have a negative TTL, and it always has a constant difference compared to the timestamp, even between different spawn points and times. So like you said, it unfortunately seems impossible to tell what the actual TTL is, which is unfortunate :(

**Comment by Chris221** [comment]
Yea there is documentation of that here too https://www.reddit.com/r/pokemongodev/comments/4wxe68/do_some_pokemon_need_a_second_pass/

**Comment by MrStu82** [comment]
I can confirm this. I've had a push bullet notifier installed in my pogomap, and I constantly get notifications for these Pokemon with negative timers. They don't show on the map and I have to use Google maps to navigate to the coords. The pokemon is always there.

As you say, they will appear for the last 15 mins.

Evidence: http://m.imgur.com/gallery/iFTbF

Pokemon didnt appear on the map til the last 15 mins.

**Comment by MrStu82** [comment]
Looking into this, I think it's a problem with how the markers are created and cleared on the map. I've just tried putting a few console breaks in there to debug. I suspect because the expiration date is in the past, the markers are instantly marked as expired and cleared from the map.

**Comment by Trippler** [comment]
> I'll post here as well. There seems to have been a bug introduced in the 595+ hour spawns fix (https://github.com/PokemonGoMap/PokemonGo-Map/commit/0f19c0a85b421a83c5c0d50c6fd39d59c1040f2c). I have a spawn that lasts for 30 minutes; and currently it shows up twice (once when it spawns, and once after 15 minutes for another 15). This is the location of the spawn point: Location: 60.368072, 5.3691274, it should spawn every hour at around XX:58 and despawn at around (XX+1):29 The notification shows up twice, and the countdown is wrong.

I assume this is the same bug. For me I'm fairly sure the pokemon show up, but count down to 0 and then show up again.

**Comment by BaIthamel** [comment]
I think this is fixed with the hack the assume ttl >15 is 15.

**Comment by sebastienvercammen** [comment]
No longer relevant due to major scan changes.
